pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -661,3 +661,13 @@ func ReadShootedSeed(shoot *gardenv1beta1.Shoot) (*ShootedSeed, error) {
 
 	return shootedSeed, nil
 }
+
+// GetKubeProxyMode returns the kube-proxy mode currently used
+func GetKubeProxyMode(shoot *gardenv1beta1.Shoot) gardenv1beta1.ProxyMode {
```

Comment:
This function is called `Get..` but it actually sets some fields in the shoot. I would not set any fields in the shoot with a function in `helper` but rather explicitly in the code where it needs to happen. Can you change?

Reply:
Had it before directly in the code, but you suggested to factor it out to helper, but will move it back.